src/osbuild-manifests/aws_tag.py

```diff
@@ -1,79 +1,95 @@
-import os
 import subprocess
 import json
 import argparse
-
-
-
+ 
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument('--stream', dest='stream', type=str, help='Fedora stream', required=True)
+    parser.add_argument('--dry-run', dest='dry_run', help='Check if the resources have tags but not add them', action='store_true')
     args = parser.parse_args()
-
+ 
     builds = getBuildsForStream(args.stream)
     for build in builds:
         build_id=build['id']
         arches=build['arches']
         for arch in arches:
-            print(f"The build is {build_id}")
+            print(f"The build is {build_id} for {arch}")
             buildFetch(args.stream, build_id, arch)
-            meta = open('builds/'+build_id+'/'+arch+'/meta.json')
+            meta = open(f'builds/{build_id}/{arch}/meta.json')
             data = json.load(meta)
-
+ 
             # Delete this when actually running. Just here while I make this script
-            data ={"amis":[{
-                "name": "us-east-1",
-                "hvm": "ami-0016d5df3041499f9",
-                "snapshot": "snap-0c1ca4850fcd5e573"
-            }]}
-            amis = data['amis']
+            # data ={"amis":[{
+            #     "name": "us-east-1",
+            #     "hvm": "ami-0016d5df3041499f9",
+            #     "snapshot": "snap-0c1ca4850fcd5e573"
+            # }]}
+            if 'amis' in data.keys():
+                amis = data['amis']
+            else:
+                print(f"{build_id} does not have any AMIs for {arch} in meta.json")
+                return

```

Comment:
we probably shouldn't return here but rather continue the loop because we want to continue to process later builds and other architectures